src/SolidClient.js

```diff
@@ -181,7 +180,22 @@ class SolidClient {
     if (this.isAboveVersion511(loginResponse.headers['x-powered-by'])) {
       const consentUrl = new URL(authUrl);
       const search = consentUrl.search.substring(1);
-      let consPostData = JSON.parse('{"' + decodeURIComponent(search).replace(/"/g, '\\"').replace(/&/g, '","').replace(/\=/g, '":"') + '"}');
+      if (!search) {
+        throw new Error(`Login response doesn't contain a search string: ${authUrl}`);
+      }
+      let consPostData = {};
+      try {
+        const searchJson = decodeURIComponent(search)
+          .replace(/"/g, '\\"')
+          .replace(/&/g, '","')
+          .replace(/\=/g, '":"');
+        consPostData = JSON.parse(`{"${searchJson}"}`);
+      }
+      catch (error) {
+        throw new Error(
+          `Login response doesn't contain a search string: ${authUrl} , cause JSON parsing error ${error}`
```

Comment:
Should be "Login response doesn't contain a search string: ${authUrl}, causing a JSON parsing error: ${error}"